util/migrate/allocator.py

```diff
@@ -1,11 +1,20 @@
 import logging
+import os
 import random
 from threading import Event
 
 from bintrees import RBTree
 
 logger = logging.getLogger(__name__)
-logger.setLevel(logging.DEBUG)  # Set the default log level to DEBUG
+
+# Read the DEBUGLOG environment variable
+debug_log = os.getenv("DEBUGLOG", "false").lower() == "true"
```

Comment:
wouldn't a boolean check be more flexible to ensure it doesn't matter how one writes `enable` ?

```
from pydantic import TypeAdatper
debug_log = TypeAdapter(bool).validate_python(
                          os.environ.get("DEBUGLOG", "False"))
```

that would give us 
```
>>> for val in (True, "True", 1, "yes", "Yes", False, "False", 0, "no", "NO"):
...        print(f"value {val} evaluated to {TypeAdapter(bool).validate_python(val)}")
... 
value True evaluated to True
value True evaluated to True
value 1 evaluated to True
value yes evaluated to True
value Yes evaluated to True
value False evaluated to False
value False evaluated to False
value 0 evaluated to False
value no evaluated to False
value NO evaluated to False
```